MerlinAU.sh

```diff
@@ -2583,36 +2583,51 @@ Please manually update to version $minimum_supported_version or higher to use th
             "$inMenuMode" && _WaitForEnterKey_ "$menuReturnPromptStr"
             return 1
         else
-            # Format current_version by removing the last '.0'
-            formatted_current_version=$(echo $current_version | awk -F. '{print $1"."$2}')
+            # Use awk to format the version based on the number of initial digits
+            formatted_current_version=$(echo "$current_version" | awk -F. '{
+                if (length($1) == 4 && NF >= 3) {
+                    # For version starting with four digits like 3004.388.5.0
+                    # Format as the next two fields (388.5)
+                    printf "%s.%s", $2, $3
+                } else if (NF >= 2) {
+                    # For version with three initial digits like 388.5.0
+                    # Format as the first two fields (388.5)
+                    printf "%s.%s", $1, $2
+                }
+            }')
 
             # Format release_version by removing the prefix '3004.' and the last '.0'
             formatted_release_version=$(echo $release_version | awk -F. '{print $2"."$3}')
 
-            # Extract log contents between two firmware versions
-            changelog_contents=$(awk "/$formatted_release_version/,/$formatted_current_version/" "$changelog_file")
-
-            # Define high-risk terms as a single string separated by '|'
-            high_risk_terms="factory default reset|features are disabled|break backward compatibility|must be manually|strongly recommended"
-
-            # Search for high-risk terms in the extracted log contents
-            if echo "$changelog_contents" | grep -Eiq "$high_risk_terms"; then
-                if [ "$inMenuMode" = true ]; then
-                    printf "\n ${REDct}Warning: Found high-risk phrases in the change-logs.${NOct}"
-                    printf "\n ${REDct}Would you like to continue anyways?${NOct}"
-                    if ! _WaitForYESorNO_ ; then
-                        Say "Exiting for change-log review."
-                        _DoCleanUp_ 1 ; return 1
+            # Check if the current version is present in the changelog
+            if ! grep -q "$formatted_current_version" "$changelog_file"; then
+                Say "Current version not found in change-log. Bypassing change-log verification for this run."
+            else
+                # Extract log contents between two firmware versions
+                changelog_contents=$(awk "/$formatted_release_version/,/$formatted_current_version/" "$changelog_file")
```

Comment:
And this:

`changelog_contents=$(awk "/^$formatted_release_version \([0-9]+[-]/,/$formatted_current_version \([0-9]+[-]/" "$changelog_file")`